x-pack/libbeat/common/aws/credentials.go

```diff
@@ -43,7 +47,19 @@ func GetAWSCredentials(config ConfigAWS) (awssdk.Config, error) {
 		return awsConfig, nil
 	}
 
-	// If accessKeyID, secretAccessKey or sessionToken is not given, then load from default config
+	// Assume IAM role if iam_role config parameter is given
+	if config.RoleArn != "" {
+		awsConfig, err := external.LoadDefaultAWSConfig()
+		if err != nil {
+			return awsConfig, err
```

Comment:
I wonder if this should be put in `errors.wrap(...)`.

Reply:
yep good point, done!